### PR TITLE
Feature - Create Todo

### PR DESCRIPTION
**Pull request by coder1033**
## Description
Add a new endpoint for creating a todo
```
URI: POST /todos
Request body: { title: … }
Response HTTP Status: 201
Response body: { id: …, title: … }
```

## Database schema changes
TodoItem model added
```
  {
    isActive: { type: Boolean, default: true },
    title: { type: String, required: true }
  }
  {
    collection: 'todos',
    timestamps: {
                 createdAt: 'createdAt',
                 updatedAt: 'updatedAt',
                }
  }
```

### Automated test cases added
- Should create a new todoitem when the valid string is passed as "title" in body, and it should return 201 status.
- Should return a validation error if an empty string is passed as "title" in body, and it should return 400 status.
- Should return a validation error if non-string is passed as "title" in body, and it should return 400 status.

### Manual test cases run
- None

### CODE REVIEW

File: test/mocks/in-memory-mongo-store.ts [hunk 1]
```diff
@@ -8,8 +8,7 @@ export class InMemoryMongoStore extends Mongoose.MongoStore {
   }
 
   public async connect(): Promise<M> {
-    const mongod = new MongoMemoryServer();
-    const uri = await mongod.getConnectionString();
+    const uri: string = config.get('mongoDb.uri');
```
Comment (vindru) [inline test/mocks/in-memory-mongo-store.ts:11]: Use mongod.getUri() here

File: src/models/todo.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { BaseModel } from './base-model';
+import { LooseObject } from '@typings';
+
+export class Todo extends BaseModel {
```
Comment (jjalan) [inline src/models/todo.ts:4]: `Todo` is a verb, not a noun. The class name should be `TodoItem`

File: src/server/app.ts [hunk 3]
```diff
@@ -89,6 +90,7 @@ export class App {
     const controllers: BaseController[] = [
       new AccountController(this.ctx),
       new HealthCheckController(this.ctx),
+      new TodoController(this.ctx),
```
Comment (jjalan) [inline src/server/app.ts:93]: `TodoItemController`

File: src/server/server.ts [hunk 4]
```diff
@@ -19,6 +19,7 @@ const respositoryContext = {
 const appContext: AppContext = {
   logger,
   accountRepository: new Repositories.AccountRepository(respositoryContext),
+  todoRepository: new Repositories.TodoRepository(respositoryContext),
```
Comment (jjalan) [inline src/server/server.ts:22]: `TodoItemRepository`

File: src/storage/mongoose/todo.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { model, Schema } from 'mongoose';
+
+const todoSchema: Schema = new Schema(
+  {
+    isActive: { type: Boolean, default: false },
```
Comment (jjalan) [inline src/storage/mongoose/todo.ts:5]: Why is default false? So you are saying any new todo item created by default is not active. That does not sound right

Comment (jjalan) [reply src/storage/mongoose/todo.ts:5]: Why have you introduced this flag Manan?

**Review by jjalan — CHANGES_REQUESTED** [review]

File: src/storage/mongoose/todo.ts (hunk 5 above)
Comment (coder1033) [reply src/storage/mongoose/todo.ts:5]: Yes, it should be true by default.
The reason for introducing this flag is that if this todo gets to be deleted then it shouldn't be removed from the database it just set as not active.

File: src/controllers/todoitem-controller.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+import { TodoItem } from '@models';
+import {
+  AppContext,
+  Errors,
+  ExtendedRequest,
+  ValidationFailure,
+} from '@typings';
+import { createTodoItemValidator } from '@validators';
+
+export class TodoItemController extends BaseController {
```
Comment (jjalan) [inline src/controllers/todoitem-controller.ts:13]: The file name should be `todo-item-controller`

File: src/models/todoitem.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import { BaseModel } from './base-model';
+import { LooseObject } from '@typings';
+
+export class TodoItem extends BaseModel {
```
Comment (jjalan) [inline src/models/todoitem.ts:4]: File name should be `todo-item`

File: src/storage/repositories/todoitem-repository.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import { BaseRepository } from './base-repository';
+import { TodoItem, ModelFactory } from '@models';
+import { RepositoryContext } from './repository-context';
+
+export class TodoItemRepository extends BaseRepository<TodoItem> {
```
Comment (jjalan) [inline src/storage/repositories/todoitem-repository.ts:5]: File name should be `todo-item-repository`

**Review by jjalan — APPROVED** [review]

**Pull request merged**